pkg/grpc/databroker/syncer.go

```diff
@@ -55,36 +54,37 @@ type Syncer struct {
 	recordVersion uint64
 	serverVersion uint64
 
-	closeOnce sync.Once
-	closed    chan struct{}
+	closeCtx       context.Context
+	closeCtxCancel func()
 }
 
 // NewSyncer creates a new Syncer.
 func NewSyncer(handler SyncerHandler, options ...SyncerOption) *Syncer {
+	closeCtx, closeCtxCancel := context.WithCancel(context.Background())
+
```

Comment:
rm and store the context in `Run` method instead

Reply:
This could lead to a nil reference exception if the syncer is closed without calling run. I can still make this change if you want but I'll have to add a mutex.